crates/sui-storage/src/mutex_table.rs

```diff
@@ -88,8 +88,15 @@ impl<K: Hash + std::cmp::Eq + Send + Sync + 'static> MutexTable<K> {
                 continue;
             }
             map.unwrap().retain(|_k, v| {
-                let mutex_guard = v.try_lock();
-                mutex_guard.is_err()
+                // (try_|)acquire_locks will lock the map and call Arc::clone on the entry
```

Comment:
Why does try_lock() lock the entire map, when `v` refers to only one Mutex?  Unless I'm misunderstanding the code here.

That said I do agree that checking for only one count/mutex copy is definitely a good optimization.
However, let's say that you check that it is 1, but somebody locks it in the meantime, then the `try_lock()` would be valuable right?

Reply:
So the comment tries to briefly explain what does `MutexMap::(try_|)acquire_locks` do (updated comment since this might not have been clear) to justify why we need to check strong count.

We **need** to check strong count here(e.g. this is not just perf optimization):
If we don't check Arc count, we might have a situation where the map is locked, new Arc is inserted in the map, and then map is unlocked and cleanup(happened to run concurrently at that time) can remove the lock due to race condition before code in `MutexMap::(try_|)acquire_locks` can acquire the lock.

Why do we not need `try_lock`(even thought I am keeping it just in case): if we were able to lock the map in the cleanup code, then one or two situations are possible
1) The map has only instance of the Arc<Mutex>, therefore checking strong_count is enough to safely delete it
2) There is some other instance of Arc<Mutex>, either transient copy in local variable in the `MutexMap::(try_|)acquire_locks`, or, owned copy of that Arc in the LockGuard(owned lock guard keeps its own Arc copy). In that case strong count will be more than 1 and our check won't pass.